tests/python/test_core.py

```diff
@@ -233,6 +233,37 @@ def test_to_dict_with_feature_metadata(self):
         dict_a_with_fm.pop('tree_data')
         self.assertEqual(obs, dict_a_with_fm)
 
+    def test_to_dict_with_metadata_nans(self):
+        nan_sample_metadata = self.sample_metadata.copy()
+        nan_feature_metadata = self.feature_metadata.copy()
+        nan_sample_metadata.at["Sample2", "Metadata4"] = np.nan
+        nan_feature_metadata.at["h", "fmdcol1"] = np.nan
+        nan_feature_metadata.at["a", "fmdcol2"] = np.nan
+
+        viz = Empress(self.tree, self.table, nan_sample_metadata,
+                      nan_feature_metadata,
+                      filter_unobserved_features_from_phylogeny=False)
+        obs = viz._to_dict()
+        dict_a_nan = copy.deepcopy(DICT_A)
+        dict_a_nan["sample_data"]["Sample2"]["Metadata4"] = np.nan
+        dict_a_nan["tip_metadata"] = {
+            "a": {"fmdcol1": "asdf", "fmdcol2": np.nan}
+        }
+        dict_a_nan["int_metadata"] = {
+            "h": {"fmdcol1": np.nan, "fmdcol2": "tyui"}
+        }
+        dict_a_nan["feature_metadata_columns"] = ["fmdcol1", "fmdcol2"]
+
+        tree_data = obs['tree_data']
+        exp = dict_a_nan['tree_data']
+        self.assert_almost_equal_tree_data(tree_data, exp)
+
+        dict_a_nan.pop('tree_data')
+        obs.pop('tree_data')
+        self.assertEqual(obs, dict_a_nan)
+
+        viz.make_empress()
```

Comment:
I suggest adding some minor checks that the template rendered correctly.

```suggestion
        res = viz.make_empress()
        
        self.assertTrue('empressRequire' in res)
        self.assertTrue('empress = new Empress' in res)
        self.assertTrue('emperor_require_logic' not in res)
```